components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.media.fullscreen
+
+import android.app.Activity
+import android.content.pm.ActivityInfo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Feature that will auto-rotate the device to the correct orientation for the media aspect ratio.
+ */
+class MediaSessionFullscreenFeature(
+    private val activity: Activity,
+    private val store: BrowserStore
+) : LifecycleAwareFeature {
+
+    private var scope: CoroutineScope? = null
+
+    @ExperimentalCoroutinesApi
+    override fun start() {
+        scope = store.flowScoped { flow ->
+            flow.map {
+                it.tabs + it.customTabs
+            }.map { tab ->
+                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
```

Comment:
I am a little uncomfortable with the !!.  However, I tried other ways but keep getting "Smart cast to 'MediaSessionState' is impossible, because 'it.mediaSessionState' is a property that has open or custom getter"